meson.build

```diff
@@ -300,10 +300,14 @@ conf.set('HAVE_LIBUDEV', lib_udev.found() ? 1 : false)
 
 lib_crypt = cc.find_library('crypt')
 
-lib_pam = cc.find_library('pam')
+req_libpam = not (get_option('build-login').disabled() or get_option('build-chfn-chsh').disabled()
+  or get_option('build-su').disabled() or get_option('build-runuser').disabled())
```

Comment:
This boolean logic here is as follows:
- create a state which is true, if any of the four are disabled, but false if all four are required/optional
- invert it
- save its value to req_libpam

This is incorrect, because it fundamentally cannot handle tristate logic, and anyways converting a feature option to a boolean means you cannot take advantage of Meson's ability to optimize a disabled feature by not doing system probes but immediately logging "dependency foo skipped: feature bar disabled".

More particularly, it means if all four feature options are auto, you want to do dependency lookups that are forgiving (not required), and if one is enabled but another is disabled you want to require the library to be found for one executable regardless of whether the other executables are not built.

Reply:
Right. This was a rudimentary hack because required expects a single feature instead of four.